provider/provider_yaml_test.go

```diff
@@ -409,3 +410,70 @@ func TestOrganizationsProjectAutoNaming(t *testing.T) {
     }
 }`)
 }
+
+func TestCheckConfigNoCredentials(t *testing.T) {
+	if !testing.Short() {
+		t.Skip("Only run in short mode, since we want to NOT have credentials.")
+	}
+	t.Setenv("GOOGLE_APPLICATION_CREDENTIALS", "")
+	t.Setenv("GOOGLE_GHA_CREDS_PATH", "")
+	t.Setenv("GOOGLE_PROJECT", "")
+	t.Setenv("GOOGLE_ZONE", "")
+	t.Setenv("GOOGLE_REGION", "")
+	credentialsValidationRun.Store(false)
```

Comment:
This is not safe, since a different run of a similar test could set `credentialsValidationRun` to `true` between this store and a call to replay. I would prefer if remove the global state here, making `credentialsValidationRun` a local variable in `Provider()`.

Reply:
I'm happy to open an issue in the bridge about creating a `PreCheckConfigCallback `interface, which will have the semantics of only ever being called once and we can schedule follow-up work on this. The issue you described is unlikely and will only effect tests, so I'd say it is out of scope for this PR.

That interface would be useful on the AWS side and then we can also carry it over to other t1 providers.

Reply:
https://github.com/pulumi/pulumi-terraform-bridge/issues/1658

Reply:
Just saw https://github.com/pulumi/pulumi-aws/pull/3310#discussion_r1470947250, good idea. Will do that instead.